python/neuroglancer/ingest/lib.py

```diff
@@ -322,169 +330,208 @@ def __repr__(self):
 
 class Bbox(object):
 
-    def __init__(self, a, b):
-        self.minpt = Vec3(
-            min(a[0], b[0]),
-            min(a[1], b[1]),
-            min(a[2], b[2])
-        )
+  def __init__(self, a, b):
+    self.minpt = Vec3(
+      min(a[0], b[0]),
+      min(a[1], b[1]),
+      min(a[2], b[2])
+    )
+
+    self.maxpt = Vec3(
+      max(a[0], b[0]),
+      max(a[1], b[1]),
+      max(a[2], b[2])
+    )
+
+  @classmethod
+  def from_vec(cls, vec):
+    return Bbox( (0,0,0), vec )
+
+  @classmethod
+  def from_filename(cls, filename):
+    match = re.search(r'(\d+)-(\d+)_(\d+)-(\d+)_(\d+)-(\d+)(?:\.gz)?$', os.path.basename(filename))
+
+    (xmin, xmax,
+     ymin, ymax,
+     zmin, zmax) = map(int, match.groups())
+
+    return Bbox( (xmin, ymin, zmin), (xmax, ymax, zmax) )
+
+  @classmethod
+  def from_slices(cls, slices3):
+    return Bbox(
+      (slices3[0].start, slices3[1].start, slices3[2].start), 
+      (slices3[0].stop, slices3[1].stop, slices3[2].stop) 
+    )
+
+  @classmethod
+  def from_list(cls, lst):
+    return Bbox( lst[:3], lst[3:6] )
+
+  def to_filename(self):
+    return '{}-{}_{}-{}_{}-{}'.format(
+      self.minpt.x, self.maxpt.x,
+      self.minpt.y, self.maxpt.y,
+      self.minpt.z, self.maxpt.z,
+    )
+
+  def to_slices(self):
+    return (
+      slice(int(self.minpt.x), int(self.maxpt.x)),
+      slice(int(self.minpt.y), int(self.maxpt.y)),
+      slice(int(self.minpt.z), int(self.maxpt.z))
+    )
+
+  def to_list(self):
+    return list(self.minpt) + list(self.maxpt)
+
+  @classmethod
+  def expand(cls, *args):
+    result = args[0].clone()
+    for bbx in args:
+      result.minpt = min2(result.minpt, bbx.minpt)
+      result.maxpt = max2(result.maxpt, bbx.maxpt)
+    return result
 
-        self.maxpt = Vec3(
-            max(a[0], b[0]),
-            max(a[1], b[1]),
-            max(a[2], b[2])
-        )
+  @classmethod
+  def clamp(cls, bbx0, bbx1):
+    result = bbx0.clone()
+    result.minpt = Vec3.clamp(bbx0.minpt, bbx1.minpt, bbx1.maxpt)
+    result.maxpt = Vec3.clamp(bbx0.maxpt, bbx1.minpt, bbx1.maxpt)
+    return result
 
-    @classmethod
-    def from_vec(cls, vec):
-      return Bbox( (0,0,0), vec )
+  def size3(self):
+    return Vec3(*(self.maxpt - self.minpt))
 
-    @classmethod
-    def from_filename(cls, filename):
-      match = re.search(r'(\d+)-(\d+)_(\d+)-(\d+)_(\d+)-(\d+)(?:\.gz)?$', os.path.basename(filename))
+  def volume(self):
+    return self.size3().rectVolume()
 
-      (xmin, xmax,
-       ymin, ymax,
-       zmin, zmax) = map(int, match.groups())
+  def center(self):
+    return (self.minpt + self.maxpt) / 2.0
 
-      return Bbox( (xmin, ymin, zmin), (xmax, ymax, zmax) )
+  def expand_to_chunk_size(self, chunk_size, offset=Vec(0,0,0, dtype=int)):
```

Comment:
`expand_to_chunk_size` is augmented to properly consider offsets. 

Previously, it was expanding to chunks that didn't exist. If you have a (64,64,64) chunk size and you input a point (5,5,5) it'll adjust it to (0,0,0), the start of the first chunk. However, if you have a dataset offset of e.g.
 (1,1,1), then (0,0,0) is not a real chunk so you have to offset it.